src/promptflow-core/promptflow/core/_serving/v2/routers/score.py

```diff
@@ -0,0 +1,114 @@
+# ---------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# ---------------------------------------------------------
+
+import asyncio
+import logging
+
+from fastapi import APIRouter, Request
+from fastapi.responses import JSONResponse
+from opentelemetry import context, trace
+from opentelemetry.trace.span import INVALID_SPAN
+
+from promptflow._utils.exception_utils import ErrorResponse
+from promptflow.contracts.run_info import Status
+from promptflow.core._serving.utils import load_request_data, try_extract_trace_context
+from promptflow.exceptions import SystemErrorException
+from promptflow.tracing._operation_context import OperationContext
+
+from ..fastapi_context_data_provider import _request_ctx_var
+from ..fastapi_response_creator import FastapiResponseCreator
+
+
+def get_score_router(logger):
+    router = APIRouter()
+
+    @router.post("/score")
+    async def score(request: Request):
+        logger.info(f"Type: {type(request.app)}, {hasattr(request.app, 'flow')}, {hasattr(request.app, 'extension')}")
+        # return {"status": "ok"}
+        app = request.app
+        raw_data = await request.body()
+        logger.debug(f"PromptFlow executor received data: {raw_data}")
+        app.init_invoker_if_not_exist()
+        if app.flow.inputs.keys().__len__() == 0:
+            data = {}
+            logger.info("Flow has no input, request data will be ignored.")
+        else:
+            logger.info("Start loading request data...")
+            data = load_request_data(app.flow, raw_data, logger)
+        # set context data
+        _request_ctx_var.get()["data"] = data
+        _request_ctx_var.get()["flow_id"] = app.flow.id or app.flow.name
+        run_id = _request_ctx_var.get().get("req_id", None)
+        # TODO: refine this once we can directly set the input/output log level to DEBUG in flow_invoker.
+        disable_data_logging = logger.level >= logging.INFO
+        span = trace.get_current_span()
+        if span == INVALID_SPAN:
+            # no parent span, try to extract trace context from request
+            logger.info("No parent span found, try to extract trace context from request.")
+            ctx = try_extract_trace_context(logger, request.headers)
+        else:
+            ctx = None
+        # TODO: change to use async once ScriptExecutor support async run
```

Comment:
I think the main branch has already supported async run?

Reply:
Let me check

Reply:
confirmed and changed to use async run for flex flow